src/Cnab.Bundle/Parameter.cs

```diff
@@ -27,42 +50,82 @@ public class ParameterMetadata
         public string Description { get; set; }
     }
 
-    public class StringParameter : Parameter
+    public class StringParameter : Parameter<string>
     {
-        [JsonProperty("defaultValue")]
-        public string DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<string> AllowedValues { get; set; }
-
         [JsonProperty("minLength")]
         public int MinimumLength { get; set; }
 
         [JsonProperty("maxLength")]
         public int MaximumLength { get; set; }
-    }
 
-    public class BoolParameter : Parameter
-    {
-        [JsonProperty("defaultValue")]
-        public bool DefaultValue { get; set; }
+        public override bool IsValid(string value)
+        {
+            if (!this.AllowedValues.Contains(value))
```

Comment:
I feel like there's a logic error here.  If `allowedValues` is not present in the JSON parameter definition, won't `this.AllowedValues` be null and you'll get a NRE?  Or if it gets deserialised as an empty list you'll still always get 'not valid', whereas in fact the intention is that there is no 'fixed' list of values.  So you need to check for null before doing this check.

Not to be snarky, but shouldn't there be tests to catch this?  I know this is early stage but if you don't unit test the foundations now then you will always be struggling to catch up.